.gitpod.yml

```diff
@@ -2,35 +2,46 @@ image:
   file: .gitpod.Dockerfile
 
 ports:
-  - port: 3000
-    onOpen: open-preview
+  - port: 3000 # client
+    onOpen: open-browser
+  - port: 5432 # postgres
+    onOpen: ignore
+  - port: 6379 # redis
+    onOpen: ignore
+  - port: 8982 # sunspot solr
+    onOpen: ignore
+  - port: 7982
+    onOpen: ignore
+
 tasks:
-  - init: >
+  - name: CircuitVerse
+    before: >
+      redis-server --daemonize yes
+    # init is not executed for prebuilt workspaces and restarts,
+    # so we should put all the heavy initialization here
+    init: >
+      echo "Installing rvm" &&
       rvm autolibs disable &&
       rvm install 2.6.5 &&
-      nvm install 12 &&
+      echo "Installing ruby bundler" &&
       gem install bundler &&
-      cp config/database.example.yml config/database.yml &&
-      createuser -s postgres &&
-      (redis-server &) &&
+      echo "Installing ruby dependencies" &&
       bundle install --without production --path vendor/bundle &&
+      echo "Installing yarn dependencies" &&
       yarn &&
+      cp config/database.example.yml config/database.yml &&
+      echo "Creating postgres user" &&
+      createuser -s postgres &&
+      echo "Setting up database" &&
       bundle exec rails db:create &&
       bundle exec rails db:schema:load &&
       bundle exec rails db:migrate &&
       bundle exec rails db:seed
     command: >
-      sleep 5 &&
-      redis-server --daemonize yes &&
-      bundle install --without production --path vendor/bundle &&
-      yarn &&
-      bundle exec rails db:create &&
-      bundle exec rails db:schema:load &&
-      bundle exec rails db:migrate &&
       bundle exec rake sunspot:solr:start &&
-      bundle exec rails db:seed &&
-      (./bin/webpack-dev-server . &) &&
-      bundle exec rails server
+      bundle exec rails server &&
+      ./bin/webpack-dev-server
```

Comment:
How will this work? Unless you start one of the processes in the background, both processes cannot run in parallel. Check if the simulator page works properly. Trying making a change and see if it is reflected.

Reply:
ahh, right!

I made small UI change to test the behavior, and found that the changes did not reflect even upon refreshing!

Locally, the changes didn't require refreshing, they were reflected immediately.

And in gitpod, after going back to how it was configured previously, the changes were reflected upon refreshing... 

so the last commit resolves the issue, but can we run one of the two command in another terminal or this way it is better?

Reply:
@tachyons @satu0king